plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/TestBigQueryParentProjectId.java

```diff
@@ -0,0 +1,94 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.bigquery;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.trino.testing.QueryRunner;
+import io.trino.testing.sql.TestTable;
+import org.junit.jupiter.api.Test;
+
+import static io.trino.testing.TestingNames.randomNameSuffix;
+import static io.trino.tpch.TpchTable.NATION;
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
+
+class TestBigQueryParentProjectId
+        extends BaseBigQueryProjectIdResolution
+{
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        queryRunner = BigQueryQueryRunner.builder()
+                .setConnectorProperties(ImmutableMap.<String, String>builder()
+                        .put("bigquery.parent-project-id", parentProjectId)
+                        .buildOrThrow())
+                .setInitialTables(ImmutableList.of(NATION))
+                .build();
+        return queryRunner;
+    }
+
+    @Test
+    void testQueriesWithParentProjectId()
+            throws Exception
+    {
+        // tpch schema is available in both projects
+        assertThat(computeScalar("SELECT name FROM bigquery.tpch.nation WHERE nationkey = 0")).isEqualTo("ALGERIA");
+        assertThat(computeScalar("SELECT * FROM TABLE(bigquery.system.query(query => 'SELECT name FROM tpch.nation WHERE nationkey = 0'))")).isEqualTo("ALGERIA");
+        assertThat(computeScalar(format("SELECT * FROM TABLE(bigquery.system.query(query => 'SELECT name FROM %s.tpch.nation WHERE nationkey = 0'))", projectId)))
+                .isEqualTo("ALGERIA");
+        assertThat(computeScalar(format("SELECT * FROM TABLE(bigquery.system.query(query => 'SELECT name FROM %s.tpch.nation WHERE nationkey = 0'))", parentProjectId)))
+                .isEqualTo("ALGERIA");
+
+        String trinoSchema = "someschema_" + randomNameSuffix();
+        try (AutoCloseable ignored = withSchema(trinoSchema); TestTable table = newTrinoTable("%s.table".formatted(trinoSchema), "(col1 INT)")) {
+            String tableName = table.getName().split("\\.")[1];
+            // schema created in parentProjectId by default
+            assertThat(computeActual(format(
+                    "SELECT * FROM TABLE(bigquery.system.query(query => 'SELECT schema_name FROM `%s.region-us.INFORMATION_SCHEMA.SCHEMATA`'))",
+                    projectId)))
+                    .doesNotContain(row(trinoSchema));
+            // confusion point: this implicitly points to project
```

Comment:
Does it point to project or parent project implicitly?

Reply:
With this configuration, it points to project ID, because it also happens to be the one set in credentials JSON. If Parent project ID is not provided, PTF calls to unprefixed datasets go to credentials JSON default project ID.

Reply:
Added more context to the comments in test to cement my understanding.